File: backend/Domain/Models/Attendance.cs
namespace backend.Domain.Models
{
    public class Attendance : BaseEntity
    {
        public int EmployeeId { get; set; }
        public Employee? Employee { get; set; }

        public DateTime Date { get; set; }

        public TimeSpan? CheckIn { get; set; }
        public TimeSpan? CheckOut { get; set; }

        public int LateMinutes { get; set; }
        public int EarlyLeaveMinutes { get; set; }
        public int WorkingMinutes { get; set; }
        public string Status { get; set; } = string.Empty;
    }
}

File: frontend/src/types/attendance.ts
// ─── Attendance ──────────────────────────────────────────
export interface AttendancePreviewDto {
  employeeId: number;
  date: string;
  checkIn: string | null;
  checkOut: string | null;
  lateMinutes: number;
  earlyLeaveMinutes: number;
  workingMinutes: number;
  status: string;
  isValid: boolean;
  errors: string[];
}

export interface SyncResult {
  inserted: number;
  updated: number;
  skipped: number;
  errors: string[];
  records: AttendanceRecordDto[];
}

export interface AttendanceRecordDto {
  employeeId: number;
  employeeName: string;
  date: string;
  checkIn: string | null;
  checkOut: string | null;
  lateMinutes: number;
  earlyLeaveMinutes: number;
  workingMinutes: number;
  status: string;
}

export interface MyAttendanceRecord {
  date: string;
  checkIn: string | null;
  checkOut: string | null;
  status: string;
  lateMinutes: number;
  earlyLeaveMinutes: number;
  workingMinutes: number;
}

export interface EmployeeMonthlyAttendanceDto {
  employeeId: number;
  employeeName: string;
  totalLateMinutes: number;
  totalEarlyMinutes: number;
  totalExcuseHours: number;
  deductionHours: number;
  status: string;
  totalWorkingHours: number;
}

export interface MyAttendanceResult {
  records: MyAttendanceRecord[];
  totalLateMinutes: number;
  totalEarlyLeaveMinutes: number;
  totalWorkingMinutes: number;
}
